src/file-upload/file-item.tsx

```diff
@@ -0,0 +1,170 @@
+import { BinIcon, CrossIcon, PencilIcon } from "@lifesg/react-icons";
+import {
+    Content,
+    DesktopErrorMessage,
+    ErrorIconButton,
+    IconButton,
+    Item,
+    ItemActionContainer,
+    ItemFileSizeText,
+    ItemNameSection,
+    ItemText,
+    LoadingActionContainer,
+    MobileErrorMessage,
+} from "./file-item.styles";
+import { FileItemProps } from "./types";
+import { ProgressBar } from "../shared/progress-bar";
+
+interface Props extends FileItemProps {
+    onDelete: () => void;
+    onEdit?: (() => void) | undefined;
+}
+
+export const FileItem = ({
+    id,
+    file,
+    description,
+    progress = 1,
+    errorMessage,
+    editableMode,
+    onDelete,
+    onEdit,
+}: Props) => {
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const isLoading = progress < 1;
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleDelete = () => {
+        onDelete();
+    };
+
+    const handleEdit = () => {
+        if (onEdit) onEdit();
+    };
+
+    // =========================================================================
+    // RENDER FUNCTIONS
+    // =========================================================================
+    const renderActionButton = () => {
+        if (isLoading) {
+            return (
+                <LoadingActionContainer>
+                    <ProgressBar progress={progress} />
+                </LoadingActionContainer>
+            );
+        } else if (errorMessage) {
+            return (
+                <ItemActionContainer>
+                    <ErrorIconButton
```

Comment:
will need `aria-label`s on these buttons and `aria-hidden` on the children. the aria-label may need to specify the file name too, as the file name doesn't get read.